portfolio/src/main/java/com/google/sps/servlets/RecognitionServlet.java

```diff
@@ -0,0 +1,55 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.util.Base64;
+
+import com.google.cloud.speech.v1.RecognitionAudio;
+import com.google.cloud.speech.v1.RecognitionConfig;
+import com.google.cloud.speech.v1.RecognizeResponse;
+import com.google.cloud.speech.v1.SpeechClient;
+import com.google.protobuf.ByteString;
+import com.google.sps.classes.AllowedLanguageCodes;
+import com.google.gson.Gson;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/speech")
+public class RecognitionServlet extends HttpServlet {
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+    String base64Audio = request.getReader().readLine();
+    byte[] decodedAudio = Base64.getDecoder().decode(base64Audio);
+    int langIndex = Integer.parseInt(request.getParameter("langIndex"));
+    int langIndexMax = AllowedLanguageCodes.getTranslationLanguageList().size() - 1;
+
+    if (langIndex > langIndexMax || langIndex < 0) {
+      response.setHeader("error", "The language sent is not currently supported.");
+      response.sendError(500);
+      return;
+    }
+
+    String languageCode = AllowedLanguageCodes.getSpeechCodeForIndex(langIndex);
```

Comment:
Since you are using the SpeechCode anyway, why not having a param that contains the speech code and not the index? All this index handling seems a bit unnecessary here.
Then you can also remove the index from the Language class

Another option: Don't call it index but id and make a function `AllowedLanguageCodes.getSpeechCodeById` that returns null if the id is unknown. So you don't need to check it here in this class.

Reply:
I use the index, because it's easier to get the translation code and speech code, since they are different. The index is just because the translation and speech APIs get the codes from the same location in the front-end.